tools/stats/stats.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __linux__
+#define _XOPEN_SOURCE 500
+#include <ftw.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <Windows.h>
+#endif
+
+
+#include <cassert>
+#include <cstring>
+#include <iostream>
+#include <unordered_map>
+
+#include "source/spirv_stats.h"
+#include "spirv-tools/libspirv.hpp"
+#include "stats_analyzer.h"
+#include "tools/io.h"
+
+using libspirv::SpirvStats;
+
+namespace {
+
+std::vector<std::string> g_input_files;
+
+void AddInputFile(const char* path) {
+  g_input_files.push_back(path);
+}
+
+#ifdef __linux__
+int AddInputFile(const char* path, const struct stat*, int tflag, struct FTW*) {
+  if (tflag == FTW_F)
+    AddInputFile(path);
+  return 0;
+}
+#endif  // __linux__
+
+bool IsDir(const char* path) {
+#ifdef __linux__
+  struct stat path_stat;
+  stat(path, &path_stat);
+  return S_ISDIR(path_stat.st_mode);
+#elif defined(_WIN32)
+  return FILE_ATTRIBUTE_DIRECTORY & GetFileAttributes(path);
+#else
+  return false;
+#endif
+}
+
+void AddInputDir(const char* path) {
+#ifdef __linux__
+  nftw(path, AddInputFile, 20, FTW_PHYS);
+#endif  // __linux__
+}
+
+void PrintUsage(char* argv0) {
+  printf(
+      R"(%s - Collect statistics from one or more SPIR-V binary file(s).
+
+USAGE: %s [options] [<filepaths>] [<dirpaths>]
```

Comment:
So this will include any regular file.  So if I dropped in a "README" somewhere it would try to be interpreted.

I would have opted to let the shell find the files for me.
For example:

```
find . -name "*.spv" -print0 | xargs -0 spirv-stats
```
Or if you're worried about command line length limits (which can be a problem on older Windows shells), then allow the user to list the files on stdin.
E.g.

```
find . -name "*.spv" | spirv-stats -
```